forge-ai/src/main/java/forge/ai/ComputerUtil.java

```diff
@@ -3257,5 +3305,12 @@ public static boolean isETBprevented(Card c) {
         List<ReplacementEffect> list = c.getGame().getReplacementHandler().getReplacementList(ReplacementType.Moved, repParams, ReplacementLayer.CantHappen);
         return !list.isEmpty();
     }
-    
+
+    public static boolean shouldSacrificeThreatenedCard(Player ai, Card c, SpellAbility sa) {
+        Game game = ai.getGame();
+        Combat combat = game.getCombat();
+        return (c.isCreature() && ComputerUtil.predictCreatureWillDieThisTurn(ai, c, sa, false)
+                && (game.getCombat() == null || !ComputerUtilCombat.willOpposingCreatureDieInCombat(ai, c, combat)))
```

Comment:
since you added null check to willOpposingCreatureDieInCombat you don't need to check it here additionally

Reply:
Nice, thanks! :) Done.